cstree/src/interning/lasso_compat/token_interner.rs

```diff
@@ -82,10 +82,12 @@ pub use multi_threaded::MultiThreadedTokenInterner;
 mod multi_threaded {
     use super::*;
 
+    use std::sync::Arc as StdArc;
+
     /// A threadsafe [`Interner`] for deduplicating [`GreenToken`](crate::green::GreenToken) strings.
     ///
-    /// Note that [`Interner`] and [`Resolver`] are also implemented for  `&MultiThreadTokenInterner` so you can pass
-    /// `&mut &interner` in shared contexts.
+    /// Note that [`Interner`] and [`Resolver`] are also implemented for `Arc<MultiThreadTokenInterner>` so you can pass
+    /// `&mut Arc::clone(interner)` in shared contexts.
```

Comment:
This one I don't think makes that much sense if you're just looking at a single method call on the interner (which this comment is about): if you have the `Arc`, you can get back to a `&mut &interner` by simply dereferencing the `Arc` (doing `&mut &*interner` instead if `interner` is an `Arc<I>`). At least, if it was my code I'd prefer doing that over cloning the `Arc`.

Maybe we should just mention the impls here without making suggestions about the code, like

```suggestion
    /// Note that [`Interner`] and [`Resolver`] are also implemented for `&MultiThreadTokenInterner` and 
    /// `Arc<MultiThreadTokenInterner>` so you can pass a mutable reference to either of these in shared contexts.
```

What do you think?

Reply:
Sounds good to me, should have considered the context when updating the doc